dagger-core/src/test/java/com/gotocompany/dagger/core/source/config/adapter/DaggerKafkaConsumerAdditionalConfigurationsAdaptorTest.java

```diff
@@ -0,0 +1,43 @@
+package com.gotocompany.dagger.core.source.config.adapter;
+
+import org.junit.Test;
+import com.google.gson.stream.JsonReader;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class DaggerKafkaConsumerAdditionalConfigurationsAdaptorTest {
+
+    @Test
+    public void shouldMapJsonStringToMap() throws IOException {
+        String input = "{\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_1\":\"value1\",\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_2\":\"value2\"}";
+        JsonReader jsonReader = new JsonReader(new StringReader(input));
+        DaggerKafkaConsumerAdditionalConfigurationsAdaptor daggerKafkaConsumerAdditionalConfigurationsAdaptor = new DaggerKafkaConsumerAdditionalConfigurationsAdaptor();
+        Map<String, String> expectedResult = new HashMap<>();
+        expectedResult.put("SOURCE_KAFKA_CONSUMER_CONFIG_KEY_1", "value1");
+        expectedResult.put("SOURCE_KAFKA_CONSUMER_CONFIG_KEY_2", "value2");
+
+        Map<String, String> result = daggerKafkaConsumerAdditionalConfigurationsAdaptor.read(jsonReader);
+
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void shouldThrowExceptionForInvalidProperties() throws IOException {
+        String input = "{\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_1\":\"value1\",\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_2\":\"value2\",\"INVALID_KEY\":\"value3\"}";
+        JsonReader jsonReader = new JsonReader(new StringReader(input));
+        DaggerKafkaConsumerAdditionalConfigurationsAdaptor daggerKafkaConsumerAdditionalConfigurationsAdaptor = new DaggerKafkaConsumerAdditionalConfigurationsAdaptor();
+
+        try {
+            daggerKafkaConsumerAdditionalConfigurationsAdaptor.read(jsonReader);
+        } catch (IllegalArgumentException e) {
+            assertEquals("Invalid additional kafka consumer configuration properties found: [INVALID_KEY]", e.getMessage());
+        }
+    }
```

Comment:
instead of try catch use expected - 
`@Test(expected = IllegalArgumentException.class)
public void shouldThrowExceptionForInvalidProperties() throws IOException {
String input = "{\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_1\":\"value1\",\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_2\":\"value2\",\"INVALID_KEY\":\"value3\"}";
JsonReader jsonReader = new JsonReader(new StringReader(input));
DaggerKafkaConsumerAdditionalConfigurationsAdaptor daggerKafkaConsumerAdditionalConfigurationsAdaptor = new DaggerKafkaConsumerAdditionalConfigurationsAdaptor();
daggerKafkaConsumerAdditionalConfigurationsAdaptor.read(jsonReader);
}`

Reply:
Intention is to assert the exception message as well, but let me update it

Reply:
@ekawinataa check if (expected = xyz, exception-message = abc) "exception-message" or some other property can be used to assert this, but if the exception message is dynamic we can't use this.

But yes, its good to assert the exception message. So verify it, if not possible keep as is.

Reply:
use rule for exception message assertions - 
```
@Rule
public ExpectedException thrown = ExpectedException.none();
@Test
public void shouldThrowExceptionForInvalidProperties() throws IOException {
String input = "{\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_1\":\"value1\",\"SOURCE_KAFKA_CONSUMER_CONFIG_KEY_2\":\"value2\",\"INVALID_KEY\":\"value3\"}";
JsonReader jsonReader = new JsonReader(new StringReader(input));
DaggerKafkaConsumerAdditionalConfigurationsAdaptor daggerKafkaConsumerAdditionalConfigurationsAdaptor = new DaggerKafkaConsumerAdditionalConfigurationsAdaptor();
thrown.expect(IllegalArgumentException.class);
thrown.expectMessage("Invalid additional kafka consumer configuration properties found: [INVALID_KEY]");
daggerKafkaConsumerAdditionalConfigurationsAdaptor.read(jsonReader);
}```

Reply:
@rajuGT @sumitaich1998 okay, I've checked the current Test annotation and there is no option to assert the message. Let's keep the message assertion, I've added explicit fail in case the expected exception is not thrown. Thanks

Reply:
no that is not the right way, please use rule to assert message like I've shown above

Reply:
updated as per discussion @sumitaich1998